src/main.ts

```diff
@@ -101,40 +101,43 @@ async function run(): Promise<void> {
     // Run gcloud versions list cmd
     await exec.exec(toolCommand, appVersionCmd, options);
 
-    const versionsToDelete = versions.slice(0, versions.length - limit);
-
-    if (versionsToDelete.length) {
+    if (versions.length > limit) {
+      const versionsToDelete = versions.slice(0, versions.length - limit);
+  
       const appDeleteCmd = [
         'app',
         'versions',
         'delete',
         ...versionsToDelete,
         '--quiet',
       ];
-
+  
       // Add gcloud flags.
       if (projectId !== '') {
         appDeleteCmd.push('--project', projectId);
       }
-
+  
       if (serviceName !== '') {
         appVersionCmd.push('--service', serviceName);
       }
-
+  
       core.debug(
         `Deleting ${
           versionsToDelete.length
         }, versions: Version ${versionsToDelete.join(' ')}`,
       );
-
+      
       // // Run gcloud cmd.
       await exec.exec(toolCommand, appDeleteCmd, options);
+      core.debug(err);
```

Comment:
Maybe keep this statement in the common flow (outside the `if`), since it might have debugging information that's useful even (or especially!) if no versions got deleted.

Reply:
Should be updated now.